auth/auth_test.go

```diff
@@ -2648,3 +2648,33 @@ func TestInvalidateChannels(t *testing.T) {
 		})
 	}
 }
+
+func TestCalculateMaxHistoryEntriesPerGrant(t *testing.T) {
+	testCases := []struct {
+		Name                     string
+		HistoryLengthInput       int
+		MaxEntriesExpectedOutput int
+	}{
+		{
+			Name:                     "Extreme history length",
+			HistoryLengthInput:       1000000,
+			MaxEntriesExpectedOutput: 1,
+		},
+		{
+			Name:                     "Large history length",
+			HistoryLengthInput:       1000,
+			MaxEntriesExpectedOutput: 10,
+		},
+		{
+			Name:                     "Tiny history length",
+			HistoryLengthInput:       1,
+			MaxEntriesExpectedOutput: 10,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			assert.Equal(t, CalculateMaxHistoryEntriesPerGrant(testCase.HistoryLengthInput), testCase.MaxEntriesExpectedOutput)
```

Comment:
expected <--> actual

```suggestion
			assert.Equal(t, testCase.MaxEntriesExpectedOutput, CalculateMaxHistoryEntriesPerGrant(testCase.HistoryLengthInput))
```